rust/chains/hyperlane-ethereum/src/contracts/mailbox.rs

```diff
@@ -357,6 +370,44 @@ where
         Ok(receipt.into())
     }
 
+    #[instrument(skip(self, messages))]
+    async fn process_batch(
+        &self,
+        messages: &[BatchItem<HyperlaneMessage>],
+    ) -> ChainResult<TxOutcome> {
+        let mut multicall = build_multicall(self.provider.clone(), &self.conn, self.domain.clone())
+            .await
+            .map_err(|e| HyperlaneEthereumError::MulticallError(e.to_string()))?;
+        let contract_call_futures = messages
+            .iter()
+            .map(|batch| async {
+                // move ownership of the batch inside the closure
+                let batch = batch.clone();
+                self.process_contract_call(
+                    &batch.data,
+                    &batch.submission_data.metadata,
+                    Some(batch.submission_data.gas_limit),
+                )
+                .await
+            })
+            .collect::<Vec<_>>();
+        let contract_calls = join_all(contract_call_futures)
+            .await
+            .into_iter()
+            .collect::<ChainResult<Vec<_>>>()?;
+
+        let batch_call = multicall::batch::<_, ()>(&mut multicall, contract_calls).await?;
+        let call = self
+            .add_gas_overrides(
+                batch_call,
+                GAS_ESTIMATE_BUFFER.checked_mul(150).map(Into::into),
```

Comment:
wonder if we should just have this be its own const to avoid the multiplication for a reader